enos/ansible/plugins/callback/influxdb_events.py

```diff
@@ -30,107 +26,129 @@ class CallbackModule(CallbackBase):
+
+        self.report_event(fields)
+
 
     def v2_playbook_on_play_start(self, play):
-        """
-        Logs each play start
-        """
+        """Log each starting play"""
 
         fields = {
-            'text': '"%s" from playbook %s started on %s by %s' %
-                (play.name, self.playbook_name, self.hostname, self.username),
+            'tags': 'play %s' % self.username,
+            'text': play.name,
+            'type': 'play',
+            'title': 'Play Started'
         }
-        self.report_event('play-start', fields)
+
+        self.report_event(fields)
+
 
     def v2_playbook_on_task_start(self, task, is_conditional):
-        """
-        Logs each task start
-        """
+        """Restart the timer when a task starts"""
+
+        self.timer = datetime.now()
+
+
+    def v2_runner_on_ok(self, result):
+        """Log each finished task marked as 'changed'"""
+
+        # Log only "changed" tasks
+        if not result.is_changed():
+            return
+
+        taskname = result._task.get_name()
+        hostname = result._host
+
+        # Record the time at the end of a task
+        end_timer = datetime.now()
+        m, s = divmod((end_timer - self.timer).seconds, 60)
+        h, m = divmod(m, 60)
+
         fields = {
-            'text': 'Task %s started on %s by %s' %
-                (task.name, self.hostname, self.username)
+            'tags': 'task %s %s %02d:%02d:%02d' % (
+                self.username,
+                hostname,
+                h, m, s),
+            'text': taskname,
+            'type': 'task',
+            'title': 'Task Succeeded'
         }
-        self.report_event('task-start', fields)
+
+        self.report_event(fields)
+
 
     def v2_playbook_on_stats(self, stats):
-        """
-        Connect to influxdb and commit events
-        """
-        try:
-            variable_manager = self.playbook._entries[0]._variable_manager
-            extra_vars = variable_manager.extra_vars
-            _host = extra_vars['influx_vip']
-        except:
-            _host = os.getenv('influx_vip')
-            if not _host:
-                self._display.warning("\
-                    Couldn't find influxdb VIP in %s playbook\n\
-                    please export the following environment\
-                    variable: 'influx_vip'" %
-                    (self.playbook_name, _host, os.path.basename(__file__)) )
-                return
+        """Connect to InfluxDB and commit events"""
+
+        # Set InfluxDB host from an environment variable if provided
+        _host = os.getenv('influx_vip') or self.extra_vars['influx_vip']
         _port = "8086"
         _user = "None"
         _pass = "None"
         _dbname = "events"
+        self.influxdb = InfluxDBClient(_host, _port, _user, _pass, _dbname)
+
         try:
-            self.influxdb = InfluxDBClient(_host, _port, _user, _pass, _dbname)
+            self.influxdb.write_points(self.events, time_precision='u')
```

Comment:
Several questions here, in case influxdb isn't reachable, 

* What will be the effect of setting `disabled=True` ? 
* I'm afraid it will block Ansible until a timeout occurs. If it's the case we should maybe make a socket test prior to sending the actual datas (Maybe the client has a built-in test ?).